bedwars-plugin/src/main/java/com/andrei1058/bedwars/support/party/FAPAdapter.java

```diff
@@ -0,0 +1,93 @@
+package com.andrei1058.bedwars.support.party;
+
+import com.andrei1058.bedwars.api.party.Party;
+import me.sk8ingduck.friendsystem.SpigotAPI;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class FAPAdapter implements Party {
+
+	private me.sk8ingduck.friendsystem.util.Party getParty(Player player) {
+		return SpigotAPI.getInstance().getMysql().getParty(player.getUniqueId());
+	}
+
+	@Override
+	public boolean hasParty(Player p) {
+		return getParty(p) != null;
+	}
+
+	@Override
+	public int partySize(Player p) {
+		me.sk8ingduck.friendsystem.util.Party party = getParty(p);
+		return party == null ? 0 : party.getAllMembers().size();
+	}
+
+	@Override
+	public boolean isOwner(Player p) {
+		me.sk8ingduck.friendsystem.util.Party party = getParty(p);
+		return party != null && party.getLeaderUUID().equals(p.getUniqueId());
+	}
+
+	public List<Player> getMembers(Player owner) {
+		me.sk8ingduck.friendsystem.util.Party party = getParty(owner);
+		return party == null ? Collections.emptyList() :
+				party.getAllMembers().stream()
+				.map(Bukkit::getPlayer)
+				.filter(Objects::nonNull)
+				.collect(Collectors.toList());
+	}
+
+	@Override
+	public void createParty(Player owner, Player... members) {
```

Comment:
if you can also complete the methods left blank and as soon as done also send a video showing that everything works we can gladly approve it.
And remember to implement support for the plugin in question on BedWarsProxy as well.

Reply:
Are the other methods necessary? Because I took a look at the [PAFBungeecordRedisApi.java](https://github.com/andrei1058/BedWars1058/blob/master/bedwars-plugin/src/main/java/com/andrei1058/bedwars/support/party/PAFBungeecordRedisApi.java) and the methods were left out there aswell. I didn't want bedwars1058 to be able to manage the parties, that's all. 
If necessary, I will implement this tho. 

And regarding BedWarsProxy, I'll make a pull request tomorrow for it aswell :)

Reply:
methods are necessary because if one uses multiArena how does one create a bedwars party using your plugin API?
if you also see in PaFSpigot the methods all are created practically and set up at the code level, as well as alessiodp's parties